### ISSUE TITLE
[BUG]It fails to read cache files with the flag_options of 'PHOTON_CONS'

### ISSUE THREAD

**Issue by jaehongpark00** [21cmfast/21cmFAST#109]
**Describe the bug:**

When I run the same script the second time, I expect the code reads in from cache instead of re-computing everything. But, with 'PHOTON_CONS', it fails with the following error message,
```
gsl: qag.c:261: ERROR: could not integrate function
Default GSL error handler invoked.
```

It happens if I exit from python (i.e. when I quit jupyter notebook and relaunch it). It happens regardless of specifying the cache folder.

Have you ever met this problem?
I attach my script.

```python
import py21cmfast as p21

p21.global_params.Z_HEAT_MAX = 30.0
p21.global_params.ZPRIME_STEP_FACTOR=1.05

flag_options = {'INHOMO_RECO': False, 'USE_TS_FLUCT': False, 'PHOTON_CONS': False,
                'SUBCELL_RSD': False, 'USE_MASS_DEPENDENT_ZETA': True}

user_params = {'DIM': 150, 'HII_DIM': 50, 'BOX_LEN': 100}
astro_params = {'F_ESC10': -1.09691, 'ALPHA_ESC': -0.4, 'M_TURN': 8.69897, 'L_X': 40.5, R_BUBBLE_MAX': 20.}

lightcone = p21.run_lightcone(
    redshift = 5.9,
    max_redshift = 15.0,
    flag_options = flag_options,
    user_params = user_params,
    astro_params = astro_params,
    direc='caches_simple',
    random_seed=999
)
```

p.s. Yuxiang tried to reproduce this problem with the above script. In his case the code recomputes everything.

**Comment by qyx268** [comment]
just want to clearify that I didn't restart the jupyter.

**Comment by andreimesinger** [comment]
@BradGreig do you mind looking into this?  I think @jaehongpark00 is stuck on the photon conservation project because of this..  many thanks

**Comment by steven-murray** [comment]
@jaehongpark00 are you using current master? Can you give a git hash?

**Comment by steven-murray** [comment]
I have reproduced this problem on current master (3d319e67c02e2a928df1fd94241767f567a92b2f).

That is, saving @jaehongpark00's script as a python file (with `PHOTON_CONS=False`), I ran it, and it ran fine. I ran it again, and everything just gets read from cache. When I ran it again with `PHOTON_CONS=True`, it fails with the integration error, after loading in the init and perturb fields from cache (those can be used from cache despite the value of `PHOTON_CONS`). 

If I then restart everything (delete everything from cache) and run with `PHOTON_CONS=True`, it runs fine. But if I then run it again (with cache), it fails with the same error.
